pkg/vpc/vpc_test.go

```diff
@@ -55,40 +352,68 @@ var _ = Describe("VPC Endpoints", func() {
 				return input != nil
 			})).Return(mockResultFn, e.Error)
 
-			if err := UseEndpointAccessFromCluster(p, cfg); err != nil {
+			if err := UseEndpointAccessFromCluster(p, e.Cfg); err != nil {
 				Expect(err.Error()).To(Equal(eks.ErrCodeResourceNotFoundException))
 			} else {
-				Expect(cfg.VPC.ClusterEndpoints.PrivateAccess).To(Equal(&e.Private))
-				Expect(cfg.VPC.ClusterEndpoints.PublicAccess).To(Equal(&e.Public))
+				Expect(e.Cfg.VPC.ClusterEndpoints.PrivateAccess).To(Equal(&e.Private))
+				Expect(e.Cfg.VPC.ClusterEndpoints.PublicAccess).To(Equal(&e.Public))
 			}
 		},
 		Entry("Private=false, Public=true", endpointAccessCase{
+			Cfg:         api.NewClusterConfig(),
 			ClusterName: "false-true-cluster",
 			Private:     false,
 			Public:      true,
 			DCOutput:    &eks.DescribeClusterOutput{Cluster: cluster},
 			Error:       nil,
 		}),
 		Entry("Private=true, Public=false", endpointAccessCase{
+			Cfg:         api.NewClusterConfig(),
 			ClusterName: "true-false-cluster",
 			Private:     true,
 			Public:      false,
 			DCOutput:    &eks.DescribeClusterOutput{Cluster: cluster},
 			Error:       nil,
 		}),
 		Entry("Private=true, Public=true", endpointAccessCase{
+			Cfg:         api.NewClusterConfig(),
 			ClusterName: "true-true-cluster",
 			Private:     true,
 			Public:      true,
 			DCOutput:    &eks.DescribeClusterOutput{Cluster: cluster},
 			Error:       nil,
 		}),
 		Entry("Private=false, Public=false", endpointAccessCase{
+			Cfg:         api.NewClusterConfig(),
 			ClusterName: "notFoundCluster",
 			Private:     false,
 			Public:      false,
 			DCOutput:    nil,
 			Error:       errors.New(eks.ErrCodeResourceNotFoundException),
 		}),
+		Entry("Nil Cluster endpoint from config", endpointAccessCase{
+			Cfg: &api.ClusterConfig{
+				TypeMeta: api.ClusterConfigTypeMeta(),
+				Metadata: &api.ClusterMeta{
+					Version: api.DefaultVersion,
+				},
+				IAM: &api.ClusterIAM{},
+				VPC: &api.ClusterVPC{
+					ClusterEndpoints: nil,
+				},
+				CloudWatch: &api.ClusterCloudWatch{
+					ClusterLogging: &api.ClusterCloudWatchLogging{},
+				},
+			},
+			ClusterName: "notFoundCluster",
+			Private:     false,
+			Public:      false,
+			DCOutput:    nil,
+			Error:       nil,
+		}),
 	)
 })
+
+func stringPointer(s string) *string {
```

Comment:
For this functionality you can use `pkg/utils/strings/strings.go:Pointer()`

Reply:
oh cool, I was trying to find the existing function for the same. Thanks for pointing out.